src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -7643,27 +7643,25 @@ private static void tryFinallyChecks(MethodHandle target, MethodHandle cleanup)
      */
     public static MethodHandle tableSwitch(MethodHandle fallback, MethodHandle... targets) {
         Objects.requireNonNull(fallback);
-        Objects.requireNonNull(targets);
-        targets = targets.clone();
-        MethodType type = tableSwitchChecks(fallback, targets);
-        return MethodHandleImpl.makeTableSwitch(type, fallback, targets);
+        var targetList = List.of(targets); // deep null checks
+        MethodType type = tableSwitchChecks(fallback, targetList);
+        return MethodHandleImpl.makeTableSwitch(type, fallback, targetList);
     }
 
-    private static MethodType tableSwitchChecks(MethodHandle defaultCase, MethodHandle[] caseActions) {
-        if (caseActions.length == 0)
-            throw new IllegalArgumentException("Not enough cases: " + Arrays.toString(caseActions));
+    private static MethodType tableSwitchChecks(MethodHandle defaultCase, List<MethodHandle> caseActions) {
+        if (caseActions.isEmpty())
+            throw new IllegalArgumentException("At least one target required");
 
         MethodType expectedType = defaultCase.type();
 
-        if (!(expectedType.parameterCount() >= 1) || expectedType.parameterType(0) != int.class)
+        if (expectedType.parameterCount() < 1 || expectedType.parameterType(0) != int.class)
             throw new IllegalArgumentException(
-                "Case actions must have int as leading parameter: " + Arrays.toString(caseActions));
+                "Switch method handles must have int as leading parameter: " + expectedType);
 
         for (MethodHandle mh : caseActions) {
-            Objects.requireNonNull(mh);
             if (mh.type() != expectedType)
                 throw new IllegalArgumentException(
-                    "Case actions must have the same type: " + Arrays.toString(caseActions));
+                    "Some targets do not have the expected type " + expectedType + ": " + caseActions);
```

Comment:
I think we should avoid changing the exception messages here, since it's observable from the outside.

FWIW, one of the downsides of only printing out a single method handle, is that a user can't tell which method handle in the list was problematic.

Reply:
At least, please motivate these changes. Why do you think changing the message is needed?

Reply:
I think changing the error message strings here is a safe move.  For a very high-use API I'd worry, a little bit, about touching the strings.  I don't mind replacing the undefined term "case action" for the term "target", which is the term used in the javadoc.  So, to use the terminology more firmly:

s/Switch method handles/Switch targets/
s/Some targets/One or more switch targets/.